<launch>

  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 1.57 0 0 /Imu_link /base_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0.493 0 0 0 0 0 /base_link /laser 100" />

  <node pkg="tf" type="static_transform_publisher" name="link3_broadcaster" args="1.85 0 0 0 0 0 /base_scanner  b_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="link3_1_broadcaster" args="-1.97 0 0 0 0.0 0 b_link camera_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="link4_broadcaster" args="0 0 0 0 0 0 camera_link openni_depth_frame  100" />

  <arg name="host" default="192.168.0.1" />
  <master auto="start"/>
  <node pkg="lms1xx" name="lms1xx" type="LMS1xx_node" output="screen">
  <param name="host" value="$(arg host)" />
  <param name="frame_id" value="base_laser" />
  </node>
  
  
  <node pkg="joy" name="joy_node" type="joy_node" output="screen">
  </node>
  
  <node pkg="donkey_rover" name="rover_odom" type="rover_odom" output="screen" required="true">
        <param name="send_odom" value="true" />
        <param name="odom_3D" value="false" />
        <param name="madgwick" value="true" />
  </node>
  
  <node type="imu_filter_node" pkg="imu_filter_madgwick" name="imu_filter_node">
    
    <param name="use_mag" value="true" />
    <param name="use_magnetic_field_msg" value="true" />
    <param name="publish_tf" value="false" />
    <param name="gain" value="0.1" />
  </node>
  
  <node pkg="gmapping" name="slam_gmapping" type="slam_gmapping" args="scan:=scan" respawn="true" output="screen">
  </node>
  
  <include file="$(find donkey_rover_2dnav)/slam/move_base_slam.launch" />
</launch>

